### PR TITLE
games-util/gamemode: Version bump

### PR DESCRIPTION
**Pull request by kakra**

### CODE REVIEW

File: games-util/gamemode/gamemode-1.4.ebuild
Comment (juippis) [inline games-util/gamemode/gamemode-1.4.ebuild]: Hmm, whats stopping from using 7?

File: games-util/gamemode/gamemode-1.4.ebuild
Comment (juippis) [inline games-util/gamemode/gamemode-1.4.ebuild]: I just wonder why this is in pkg_pretend, no one will have time to read it before it starts compiling so doesnt matter if its in here or pkg_postinst for example. **If** you need to check whether that kernel option is enabled, there are linux-* eclasses to do so. But seems like its not the case, after reading the message.

File: games-util/gamemode/gamemode-1.4.ebuild
Comment (juippis) [inline games-util/gamemode/gamemode-1.4.ebuild]: `local emesonargs=(`

File: games-util/gamemode/gamemode-1.4.ebuild
Comment (juippis) [inline games-util/gamemode/gamemode-1.4.ebuild]: ... and you can drop everything after `meson_src_configure`.

File: games-util/gamemode/gamemode-1.4.ebuild
Comment (juippis) [inline games-util/gamemode/gamemode-1.4.ebuild]: This needs the new acct-group method of handling groups :\

These comments apply to -9999 as well.

Comment (kakra) [reply games-util/gamemode/gamemode-1.4.ebuild]: Does it need EAPI=7 also in that case?

File: games-util/gamemode/gamemode-1.4.ebuild
Comment (kakra) [reply games-util/gamemode/gamemode-1.4.ebuild]: I was unsure about profile compatibility, so I didn't bump that to 7. But I can do so.

File: games-util/gamemode/gamemode-1.4.ebuild
Comment (kakra) [reply games-util/gamemode/gamemode-1.4.ebuild]: I was just replicating what a lot of other packages do when checking for recommended kernel options: They do that before merging starts. And I always concluded that's because it leaves the opportunity to cancel emerge now and adjust those things, or decide against using it at all then.

I'll move that to postinst if you still think it makes more sense.

File: games-util/gamemode/gamemode-1.4.ebuild
Comment (juippis) [reply games-util/gamemode/gamemode-1.4.ebuild]: It shouldn't, but I haven't tested and verified that.

File: games-util/gamemode/gamemode-1.4.ebuild
Comment (juippis) [reply games-util/gamemode/gamemode-1.4.ebuild]: Sure sure, it's fine.

**Comment by juippis** [comment]
And you'll need to reserve the GID by sending a mail to gentoo-dev mailing list. For example I see 405 UID is being reserved, so maybe some other GID would suit better? Whatever you choose, you should check that Arch and Fedora haven't reserved that GID either to stay compatible.

**Comment by kakra** [comment]
Actually, I've sent such a mail but looking at the list archives now, I see it doesn't show up there (but I didn't receive the NDR either).

Before I resent the mail, let's check https://api.gentoo.org/uid-gid.txt: According to this list, GID 405 is not in use (there's a UID 405 which has no GID associated). Does the dash mean that the GID is also reserved?

Where do I look up the assignments of Fedora or Arch? Do they have a similar list?

**Comment by kakra** [comment]
@juippis https://archives.gentoo.org/gentoo-dev/message/06b4c45858a15e78cb44efd6cf679600

**Comment by juippis** [comment]
> Before I resent the mail, let's check https://api.gentoo.org/uid-gid.txt: According to this list, GID 405 is not in use (there's a UID 405 which has no GID associated). Does the dash mean that the GID is also reserved?

It's not, but since UID is, I'd like to leave 405 possibility for its GID as well. Also if you ever need UID for your package, you wont have to find something else than 405.

File: games-util/gamemode/gamemode-1.5.1.ebuild
Comment (juippis) [inline games-util/gamemode/gamemode-1.5.1.ebuild:28]: https://github.com/FeralInteractive/gamemode/blob/master/meson_options.txt systemd looks like an optional dependency.

File: games-util/gamemode/gamemode-1.5.1.ebuild
Comment (juippis) [inline games-util/gamemode/gamemode-1.5.1.ebuild:6]: This is probably not needed. Are you sure multilib does anything with this ebuild? Just by looking at the ebuild, I have doubts (can't test yet due to systemd dep)

File: games-util/gamemode/gamemode-1.5.1.ebuild
Comment (juippis) [inline games-util/gamemode/gamemode-1.5.1.ebuild:97]: These need to be appended with `|| die`.

File: games-util/gamemode/gamemode-1.5.1.ebuild
Comment (juippis) [inline games-util/gamemode/gamemode-1.5.1.ebuild:72]: I think this makes it search for libsystemd. You should try to make it a conditional dep.

```
Run-time dependency libsystemd found: NO (tried pkgconfig and cmake)

meson.build:83:0: ERROR: Dependency "libsystemd" not found, tried pkgconfig and cmake

A full log can be found at /tmp/portage/games-util/gamemode-1.5.1/work/gamemode-1.5.1-abi_x86_32.x86/meson-logs/meson-log.txt
 * ERROR: games-util/gamemode-1.5.1::testworld failed (configure phase):
 *   (no error message)
```

File: games-util/gamemode/gamemode-1.5.1.ebuild
Comment (juippis) [inline games-util/gamemode/gamemode-1.5.1.ebuild:28]: I wonder would it look better to depend either on dbus, or systemd?
If that's not wise, just make systemd a conditional dep and let's try with that. 

Or if you're 100 % sure it doesn't work at all without it, then unconditional dependency on systemd is ok I guess.

Are you sure the 32-bit lib needs 32-bit libsystemd?

Comment (kakra) [reply games-util/gamemode/gamemode-1.5.1.ebuild:28]: You're probably true: depending on either dbus or systemd should cover everything. Also, I think it actually doesn't need the multilib depend because only the daemon itself (64-bit) links to libsystemd.so, the 32-bit libs don't. But they link to 32-bit dbus, so we'd need a multilib dep on that?

File: games-util/gamemode/gamemode-1.5.1.ebuild
Comment (kakra) [reply games-util/gamemode/gamemode-1.5.1.ebuild:72]: Yeah, let's make a useflag then. We are also linking to systemd in one of the libs, so switching systemd should rebuild this package.

File: games-util/gamemode/gamemode-1.5.1.ebuild
Comment (juippis) [inline games-util/gamemode/gamemode-1.5.1.ebuild:72]: This still doesn't work for me :\
```
 * abi_x86_32.x86: running multilib-minimal_abi_src_configure
meson setup --buildtype plain --libdir lib --localstatedir /var/lib --prefix /usr --sysconfdir /etc --wrap-mode nodownload --cross-file /tmp/portage/games-util/gamemode-1.5.1/temp/meson.i686-pc-linux-gnu.x86 -Dwith-systemd=false -Dwith-systemd-user-unit-dir=/usr/lib/systemd/user -Dwith-examples=false -Dwith-daemon=false /tmp/portage/games-util/gamemode-1.5.1/work/gamemode-1.5.1 /tmp/portage/games-util/gamemode-1.5.1/work/gamemode-1.5.1-abi_x86_32.x86
...
...
Run-time dependency libsystemd found: NO (tried pkgconfig and cmake)
meson.build:83:0: ERROR: Dependency "libsystemd" not found, tried pkgconfig and cmake
```

Weird that they have a systemd flag if it's not optional after all...?

File: acct-group/gamemode/metadata.xml
Comment (juippis) [inline acct-group/gamemode/metadata.xml:5]: (Is this still your e-mail address you use in bugzilla?)

Comment (kakra) [reply acct-group/gamemode/metadata.xml:5]: Yes... It's the only bugzilla account I have. But I use different addresses for "Signed-off-by" and authorship in Git. Is that a problem?

File: games-util/gamemode/gamemode-1.5.1.ebuild
Comment (kakra) [reply games-util/gamemode/gamemode-1.5.1.ebuild:72]: Hmm, yes... I'll report upstream. Does it work correctly for you if you don't specify `--with-systemd` at all? Maybe I should setup a container without systemd, tho that's not strictly supported with systemd containers (but it kinda works).

Comment (juippis) [reply games-util/gamemode/gamemode-1.5.1.ebuild:72]: Nope, I don't really have the time to dig into their meson.build but I'm sure there's some sort of small mistake that could be patched. However if it looks like a huge task, then it's better to make this depend on systemd for now I guess.

File: acct-group/gamemode/metadata.xml
Comment (juippis) [reply acct-group/gamemode/metadata.xml:5]: It's not a problem, for some reason that e-mail ringed a bell and I wanted to make sure this wasn't the reason.

There were some maintainers who had switched their bugzilla e-mails while leaving old ones to metadata.xml -> bugs can't be assigned.

Comment (kakra) [reply acct-group/gamemode/metadata.xml:5]: But your bell worked correctly, we already discussed this topic. ;-)

File: games-util/gamemode/gamemode-1.5.1.ebuild
Comment (kakra) [reply games-util/gamemode/gamemode-1.5.1.ebuild:72]: Okay, I'll let it depend on systemd for now and work with upstream on fixing it. Next revbump should then support non-systemd properly.

Comment (kakra) [reply games-util/gamemode/gamemode-1.5.1.ebuild]: I've files a bug report upstream and temporarily worked around it by setting `REQUIRED_USE`, hope that's ok.

File: games-util/gamemode/gamemode-9999.ebuild
Comment (kakra) [inline games-util/gamemode/gamemode-9999.ebuild:28]: @juippis Is this okay as a provisional work-around?

**Comment by kakra** [comment]
> There's no need to make it so complicated :) Just hard-depend on systemd explaining that's how upstream does it now while adding a link to the reported issue so it can be tracked.

Actually, I wanted to keep it easy for reverting the change... ;-) Currently, I'd just need to remove the `REQUIRED_USE` line - or did I miss something?

**Comment by kakra** [comment]
@juippis Can we expect a non-systemd Gentoo to have elogind? The latter should also come with `sd-bus.h` which is the interface needed by gamemoded. It just uses that dbus header, it doesn't actually depend on any systemd specific implementation. If you're having elogind, I'd be interested if it installed `sd-bus.h`, and into which location.

**Comment by juippis** [comment]
Currently no. You can have none, consolekit, elogind or systemd.

Gentoo is heading to a direction where elogind would become the default when not systemd, but currently it's not.

File: games-util/gamemode/gamemode-1.5.1.ebuild
Comment (juippis) [inline games-util/gamemode/gamemode-1.5.1.ebuild:28]: You can just use `REQUIRED_USE="systemd"`

**Comment by kakra** [comment]
The source code will need some implementation of `sd-bus.h`, it should probably work if systemd or elogind is just present so it can link the sd-bus implementation at runtime. It won't work without dbus, tho. So what would you suggest? Depend on either systemd or elogind, or just statically on systemd? Upstream already has a patch prepared which should accept both.

**Comment by juippis** [comment]
Just hard-depend on dbus then. It's fine. 

Supporting both elogind and systemd would be the best, I have a feeling the userbase is pretty 50-50 split.

File: games-util/gamemode/gamemode-1.5.1.ebuild
Comment (kakra) [inline games-util/gamemode/gamemode-1.5.1.ebuild]: Hmm, I think we can drop this? The usedep on dbus should pull it in, shouldn't it?

Comment (kakra) [reply games-util/gamemode/gamemode-1.5.1.ebuild]: I just dropped it.

**Comment by kakra** [comment]
> Currently no. You can have none, consolekit, elogind or systemd.
> 
> Gentoo is heading to a direction where elogind would become the default when not systemd, but currently it's not.

At a first glace, it seems like it's default for the desktop profile:
```
# fgrep elogind -r /usr/portage/profiles/
[...]
/usr/portage/profiles/targets/desktop/make.defaults:USE="a52 aac acpi alsa bluetooth branding cairo cdda cdr cups dbus dri dts dvd dvdr elogind emboss encode exif flac gif gpm gtk icu jpeg lcms ldap libnotify mad mng mp3 mp4 mpeg ogg opengl pango pdf png policykit ppds qt5 sdl spell startup-notification svg tiff truetype vorbis udev udisks unicode upower usb wxwidgets X xcb x264 xml xv xvid"
```

**Comment by juippis** [comment]
```
>>> Configuring source in /tmp/portage/games-util/gamemode-1.5.1/work/gamemode-1.5.1 ...
 * abi_x86_32.x86: running multilib-minimal_abi_src_configure
meson setup --buildtype plain --libdir lib --localstatedir /var/lib --prefix /usr --sysconfdir /etc --wrap-mode nodownload --pkg-config-path=/usr/share/pkgconfig --cross-file /tmp/portage/games-util/gamemode-1.5.1/temp/meson.i686-pc-linux-gnu.x86 -Dsystemd=false -Delogind=true -Dwith-systemd-user-unit-dir=/usr/lib/systemd/user -Dwith-examples=false -Dwith-daemon=false --wrap-mode forcefallback /tmp/portage/games-util/gamemode-1.5.1/work/gamemode-1.5.1 /tmp/portage/games-util/gamemode-1.5.1/work/gamemode-1.5.1-abi_x86_32.x86
...
...
...
Run-time dependency libsystemd found: NO (tried pkgconfig and cmake)

meson.build:90:4: ERROR: Dependency "libsystemd" not found, tried pkgconfig and cmake
```

**Comment by kakra** [comment]
Hmm, the patch should have fixed it... Did the patch from files dir apply?

**Comment by kakra** [comment]
@juippis Please try again, `meson_use` does not add the `with-` prefix to the feature name.

**Comment by juippis** [comment]
```
Run-time dependency libelogind found: NO (tried pkgconfig and cmake)

meson.build:93:4: ERROR: Dependency "libelogind" not found, tried pkgconfig and cmake
```
So this is now a multilib problem... works fine with `ABI_X86="-32"`

**Comment by kakra** [comment]
> So this is now a multilib problem... works fine with `ABI_X86="-32"`

Should be fixed now with the latest patch by upstream... It no longer tests against the sd-bus implementation when building without the daemon.

Sorry for all the trouble, I hope this is the last time I have to ask: Does it work for you now?

**Comment by kakra** [comment]
Changed to "draft" because current patches no longer build the daemon...

**Comment by juippis** [comment]
Don't worry about, I'd much rather have this be a top-notch quality even if it takes some time and testing. 

Thanks for delving into it. Let me know when you consider it finished.

**Comment by MathyV** [comment]
I think you used an older version of the patch while you were testing. The commits that finally got into master fix the typo that prevents the daemon from being built:

https://github.com/FeralInteractive/gamemode/commit/953792b4a54f1a05bda1bb98e8ec65fcff455386
https://github.com/FeralInteractive/gamemode/commit/ce6485ef97239c06b606585698defad00e25ad19

With this, the daemon is built for me. However, this issue hasn't been resolved yet:

https://github.com/FeralInteractive/gamemode/issues/222

(I can see how you thought it was, but I guees that was just because the daemon wasn't building).

I had to build amd64 only to be able to get it to pass the QA and install. Wouldn't the easiest solution be to create a small package for the inih? I did something similar with the stb package when Cura started requiring that.

I didn't want to just open a new PR and steal your thunder since you've put so much effort into it already :-) Let me know if you need help with getting this PR done, I have some time to pick it up if you want.

**Comment by kakra** [comment]
> I think you used an older version of the patch while you were testing. The commits that finally got into master fix the typo that prevents the daemon from being built:

Yes, I noticed that.

> I had to build amd64 only to be able to get it to pass the QA and install. Wouldn't the easiest solution be to create a small package for the inih? I did something similar with the stb package when Cura started requiring that.
> 
> I didn't want to just open a new PR and steal your thunder since you've put so much effort into it already :-) Let me know if you need help with getting this PR done, I have some time to pick it up if you want.

I'm currently lacking some time but I'll get back to this shortly.

If you'd like to help me out with the inih package, I'd happily pick that up as a dependency of this package. I just didn't want to create another package to maintain. Then we can finally and quite easily get this fixed.

**Comment by WillPower3309** [comment]
looks like version 1.6 supports elogind
https://github.com/FeralInteractive/gamemode/releases/tag/1.6

**Review by kakra — COMMENTED** [review]
Introduces new dependency dev-libs/inih to fix git subproject issues and avoid static linking, also bumps to current gamemode 1.6 which introduces the new dependencies and fixes elogind compatibility. Please review.

**Comment by kakra** [comment]
@juippis Please test again. I'm not sure if we need `MULTILIB_USEDEP` on `dev-libs/inih` but I don't think so because only the single-arch daemon links to it, the 32-bit libs should be fine without it.

File: acct-group/gamemode/gamemode-0.ebuild
Comment (kakra) [inline acct-group/gamemode/gamemode-0.ebuild:9]: Switched to 501 according to https://wiki.gentoo.org/wiki/Proxied_Maintainer_FAQ#Adding_acct-.7Bgroup.2Cuser.7D.2Fpackages_as_a_proxied_maintainer

**Review by kakra — COMMENTED** [review]
Changed to GID 501, added a close tag for bugs.gentoo.org

File: acct-group/gamemode/gamemode-0.ebuild
Comment (kakra) [inline acct-group/gamemode/gamemode-0.ebuild:9]: Next free GID downwards from 500 is 385 as of commit time

**Review by juippis — APPROVED** [review]
Seems to work, thanks for patience and thorough work here :) 

I'll fix a typo I spotted while at it.

**Comment by kakra** [comment]
Typos are owned by the proof reader - feel free. :-)

**Pull request closed without merging**
